File: Practica 8/Problema2/Dart/Mammal.dart
import 'AbstractAnimal.dart';

class Mammal extends AbstractAnimal {
  Mammal(String name, int year) : super.namedYear(name, year);

  @override
  String move() {
    return "walk";
  }

  @override
  String breathe() {
    return "lungs";
  }

  @override
  String reproduce() {
    return "live births";
  }
}

File: Practica 8/Problema2/Kotlin/Mammal.kt
import AbstractAnimal

class Mammal(name: String, year: Int) : AbstractAnimal(name, year) {
    override fun move(): String {
        return "walk"
    }

    override fun breathe(): String {
        return "lungs"
    }

    override fun reproduce(): String {
        return "live births"
    }

    override fun getName(): String {
        return super.getName()
    }
}
